src/odemis/model/_vattributes.py

```diff
@@ -990,22 +990,31 @@ def clip(self, val):
         clipped separately by position for positions 0..n-1
         return (same type as val): value clipped
         """
+        if val in self.choices:
+            return val
+
+        def calcDistance(choice):
+            """Calculate euclidean distance. If choice contains non numbers,
+            an infinite distance is returned."""
+            try:
+                ls.append((choice, distance.euclidean(choice, val)))
+            except Exception:
+                pass
+
+        # find the closest choice (for numbers or tuples only)
+        if isinstance(val, tuple) or isinstance(val, numbers.Real):
```

Comment:
You could make it a tiny bit more generic by checking for `collections.Iterable` instead of `tuple`.